stable/database/files/nuosm

```diff
@@ -511,11 +531,10 @@ if [ "$SNAPSHOT_RESTORED" == "true" ]; then
   loadFromSnapshot
 fi
 
-# Retry the first request against the admin layer to make sure that Raft
-# commands can be committed and fail fast on error
-retry 3 nuocmd get value --key "$liveness_req" 2>&1
-if [ $? -ne 0 ]; then
-  die 1 "NuoDB Admin layer liveness check failed"
+# Perform a consistent read of an arbitrary KV-store entry in a polling loop to
+# confirm Raft leadership before proceeding with SM start-up.
+if ! retry 60 nuocmd get value --key nuodb/nuosm/readiness; then
```

Comment:
Nit: In the ds-nuodb component, I am using nuocmd check servers:

status="$(nuocmd check servers --check-leader --timeout "$timeout" 2>&1)"
  if [ $? -ne 0 ]; then
    die 1 "Admin layer is inoperative - exiting: ${status}"
  fi

In general, I try to use the timeouts built into nuocmd/nuodocker rather than a polling loop in the script.

Since retry() is used in multiple places, I'm not sure it's worth changing it in this PR.

Reply:
Agreed, but I tend to be conservative with changes that can have a big impact. This code path is exercised every time an SM starts up, so I didn't want to introduce too many changes in behavior.